src/cider/nrepl/middleware/out.clj

```diff
@@ -64,19 +63,40 @@
                       (.flush printer))))
                 true))
 
+(defn print-stream
+  "Returns a PrintStream suitable for binding as java.lang.System/out
+  or java.lang.System/err. All operations are forwarded to all output
+  bindings in the sessions of messages in addition to the server's
+  usual PrintWriter (saved in `original-out` or `original-err`).
+  type is either :out or :err."
+  [type]
+  (let [printer (case type
+                  :out '*out*
+                  :err '*err*)]
+    (PrintStream. (proxy [OutputStream] []
+                    (close [] (.flush ^OutputStream this))
+                    (write
+                      ([b]
+                       (.write @(resolve printer) (String. b)))
+                      ([b ^Integer off ^Integer len]
+                       (.write @(resolve printer) (String. b) off len)))
+                    (flush []
+                      (.flush @(resolve printer))))
+                  true)))
+
 ;;; Known eval sessions
 (def tracked-sessions-map
   "Map from session ids to eval `*msg*`s.
   Only the most recent message from each session is stored."
   (atom {}))
 
 (defn tracked-sessions-map-watch [_ _ _ new-state]
-  (let [o (forking-printer (vals new-state) :out)]
-    ;; FIXME: This won't apply to Java loggers unless we also
-    ;; `setOut`, but for that we need to convert a `PrintWriter` to a
-    ;; `PrintStream` (or maybe just not use a `PrintWriter` above).
-    ;; (System/setOut (PrintStream. o))
-    (alter-var-root #'*out* (constantly o))))
+  (let [ow (forking-printer (vals new-state) :out)
```

Comment:
haha, okay.  I was trying to follow the existing conventions, which seemed to favor short names in this file.  Plus, `ow` and `ew` made me chuckle.  I use more descriptive names.